grammar.js

```diff
@@ -246,28 +295,40 @@ module.exports = grammar(require('tree-sitter-javascript/grammar'), {
       repeat(
         choice(
           seq($.method_definition, optional(';')),
-          $.ambient_method_declaration,
+          $.abstract_method_definition,
           seq($.public_field_definition, semicolon($)),
           seq($.index_signature, semicolon($))
         )
       ),
       '}'
     ),
 
-    method_definition: $ => seq(
-      optional($.accessibility_modifier),
-      optional('static'),
-      optional($.readonly),
-      optional('async'),
-      optional(choice('get', 'set', '*')),
-      $._property_name,
-      $.call_signature,
-      $.statement_block
-    ),
+    method_definition: $ => prec.left(1, choice(
+      seq(
+        optional($.accessibility_modifier),
+        optional('static'),
+        optional($.readonly),
+        optional($.reserved_identifier),
+        optional(choice('get', 'set', '*')),
+        $._property_name,
+        $.call_signature,
+        $.statement_block
+      ),
+      seq(
+        optional($.accessibility_modifier),
+        optional('static'),
+        optional($.readonly),
+        optional($.reserved_identifier),
+        optional(choice('get', 'set', '*')),
+        $._property_name,
+        $.call_signature
+      )
+    )),
```

Comment:
Just for conciseness, could we collapse the two choices into one, with an `optional($.statement_block)` at the end?